control-plane/agents/core/src/node/registry.rs

```diff
@@ -17,7 +17,10 @@ impl Registry {
         let nodes = self.nodes().read().await;
         let mut nodes_vec = vec![];
         for node in nodes.values() {
-            nodes_vec.push(node.read().await.node_state().clone());
+            let node_id = node.read().await.id().clone();
```

Comment:
can you get the `node_id` from the "copy_node_state" ?

Reply:
You're right. Fixing.